### PR TITLE
New global loading on all pages

### PR DESCRIPTION
**Pull request by batpad**
@frozenhelium added the new global loading to accounts and deployments pages.

Could you please give this a quick review and if it looks okay, I can proceed with implementing this across all the pages where we currently use the old loading pattern.

Thanks!

cc @GregoryHorvath

### CODE REVIEW

File: src/root/views/deployments.js [hunk 1]
```diff
@@ -267,6 +265,12 @@ class Deployments extends SFPComponent {
 
   render () {
     const { strings } = this.context;
+    const pending = this.isPending(this.props.eruOwners);
```
Comment (frozenhelium) [inline src/root/views/deployments.js:268]: We can directly do
```
const pending = this.props.eruOwners?.fetching
```

File: src/root/views/deployments.js [hunk 2]
```diff
@@ -267,6 +265,12 @@ class Deployments extends SFPComponent {
...
+    if (pending) {
```
Comment (frozenhelium) [inline src/root/views/deployments.js:269]: I think we can directly do
```
<App className='page--deployments'>
    { pending && <NewGlobalLoading /> }
    ...
```

File: src/root/views/account.js [hunk 3]
```diff
@@ -791,6 +792,12 @@ class Account extends React.Component {
 
   render () {
     const { strings } = this.context;
+    const pending = this.isPending(this.props.profile);
+    if (pending) {
+      return (
+        <NewGlobalLoading />
+      );
+    }
```
Comment (frozenhelium) [inline src/root/views/account.js:800]: we can figure out pending inline as
```
const pending = this.props.profile?.fetching;
```

And put the global loading without blocking the view i.e. inside `App`
```
<App className='page--account'>
    { pending && <NewGlobalLoading /> }
    ...
```

File: src/root/views/deployments.js (hunk 2 above)
Comment (batpad) [reply src/root/views/deployments.js:269]: So the problem with that was that the rest of the render depends on variables that are not available yet and that causes errors. I could go through on each page and make sure that things don't error, but that seems a bit brittle, and in some cases it seems much safer to short-circuit the return and not attempt to render the main view. Any suggestions here for doing this better? This didn't seem great to me, but could not figure a better way without getting deep into each view to ensure it doesn't error if data isn't available yet.

File: src/root/views/account.js (hunk 3 above)
Comment (batpad) [reply src/root/views/account.js:800]: Same as above.

File: src/root/views/deployments.js (hunk 2 above)
Comment (frozenhelium) [reply src/root/views/deployments.js:264]: So for that what we can do is conditionally call the `renderContent()`. Something like
```jsx
<App className='page--deployments'>
    ...
    { this.props.eruOwners?.fetched && this.renderContent() }
```

Comment (frozenhelium) [reply src/root/views/deployments.js:264]: The reason we want to render `App` is that it'll show the headers. If we don't it'll look similar to how emergencies page currently load (i.e. without header / navbar)

**Comment by batpad** [comment]
@frozenhelium we did not manage to land this :( - hopefully this can be worked on as part of request and loading refactors, but closing this to clean up.

**Pull request closed without merging**